scripts/uploader.rb

```diff
@@ -35,46 +34,174 @@ def run
         end
 
         destination_name = uploaded_filename(appliance)
-        source_name = File.expand_path(appliance)
+        source_name      = File.expand_path(appliance)
+        upload_options   = {:source_hash => Digest::MD5.file(source_name).hexdigest}
 
-        puts "Uploading #{appliance} as #{destination_name}..."
-
-        destination_url = url(destination_name)
-        source_hash = Digest::MD5.file(source_name).hexdigest
-
-        upload_headers = headers.merge("ETag" => source_hash)
         if nightly?
           image_date = destination_name.match(/.*([0-9]{8}).*/)[1]
           delete_at  = (DateTime.parse(image_date) + NIGHTLY_BUILD_RETENTION_TIME)
-          upload_headers["X-Delete-At"] = delete_at.to_i.to_s
+          upload_options[:expires] = delete_at
+        end
+
+        digital_ocean_client.upload(source_name, destination_name, upload_options)
+        rackspace_client.upload(source_name, destination_name, upload_options)
```

Comment:
And this can be `clients.each { |c| c.upload(...) }`

Reply:
Done